src/test/java/uk/gov/justice/digital/delius/controller/secure/OffenderResource_CheckUserAccess.java

```diff
@@ -0,0 +1,138 @@
+    private final NsiService nsiService = mock(NsiService.class);
+    private final SentenceService sentenceService = mock(SentenceService.class);
+    private final UserService userService = mock(UserService.class);
+    private final CurrentUserSupplier currentUserSupplier = mock(CurrentUserSupplier.class);
+
+    @BeforeEach
+    public void setup() {
+        RestAssuredMockMvc.standaloneSetup(
+                new OffendersResource(offenderService, alfrescoService, documentService, contactService, convictionService, nsiService, offenderManagerService, sentenceService, userService, currentUserSupplier),
+                new SecureControllerAdvice()
+        );
+    }
+
+    @Test
+    void willCheckUserAccessIsNotRestrictedOrExcluded(){
+        final var offender = theOffender();
+        given(offenderService.getOffenderByCrn(SOME_CRN_NUMBER)).willReturn(offender);
+        given(currentUserSupplier.username()).willReturn(Optional.of("BOB"));
+        given(userService.accessLimitationOf("BOB", offender.get())).willReturn(accessLimitationNone());
+
+        final var accessLimitation =  given()
+                .contentType(APPLICATION_JSON_VALUE)
+                .when()
+                .get(String.format("/secure/offenders/crn/%s/userAccess", SOME_CRN_NUMBER))
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .as(AccessLimitation.class);
+
+        assertThat(accessLimitation.isUserExcluded()).isFalse();
+        assertThat(accessLimitation.getExclusionMessage()).isNullOrEmpty();
+        assertThat(accessLimitation.isUserRestricted()).isFalse();
+        assertThat(accessLimitation.getRestrictionMessage()).isNullOrEmpty();
+
+    }
+
+    @Test
+    void willCheckUserAccessIsRestrictedAndExcluded(){
+        final var offender = theOffender();
+        given(offenderService.getOffenderByCrn(SOME_CRN_NUMBER)).willReturn(offender);
+        given(currentUserSupplier.username()).willReturn(Optional.of("BOB"));
+        given(userService.accessLimitationOf("BOB", offender.get())).willReturn(accessLimitationRestrictedAndExcluded());
+
+        final var accessLimitation =  given()
+                .contentType(APPLICATION_JSON_VALUE)
+                .when()
+                .get(String.format("/secure/offenders/crn/%s/userAccess", SOME_CRN_NUMBER))
+                .then()
+                .statusCode(403)
+                .extract()
+                .body()
+                .as(AccessLimitation.class);
+
+        assertThat(accessLimitation.isUserExcluded()).isTrue();
+        assertThat(accessLimitation.getExclusionMessage()).isEqualTo("Your excluded");
+        assertThat(accessLimitation.isUserRestricted()).isTrue();
+        assertThat(accessLimitation.getRestrictionMessage()).isEqualTo("Your Restricted");
+
+
+    }    @Test
+    void willCheckUserAccessOffenderNotFound(){
```

Comment:
The swagger doc mentioned a not found for unknown user too - does that need a test?

Reply:
Sorry ignore me - just found that one in the integration test :smile:

Reply:
Ignored !